### PR TITLE
Refactor Testing: Unused Variables, Imprecise Asserts

### PR DESCRIPTION
**Pull request by dyastremsky**
A small refactor of some of the testing and models to get rid of unused variables, imprecise asserts, unnecessary imports, and non-recommended syntax.

### CODE REVIEW

**Comment by tanmayv25** [comment]
Looks good to me. But @GuanLuo can review mlflow side changes. 

@dyastremsky  Did you use any tool to detect the issues? Can you do something similar for Python backend?

**Comment by dyastremsky** [comment]
@tanmayv25 Yup! I saw someone put our repo into LGTM, so I refactored some of the higher-significance alerts. Just added the python_backend repo, it might take some time to analyze.

LGTM link here: https://lgtm.com/projects/g/triton-inference-server/server/

File: qa/common/nightly_email_helper.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 def send(subject: str,
          content: str,
-         attachments=[],
```
Comment (CoderHam) [inline qa/common/nightly_email_helper.py:42]: Why is this change needed?

**Review by CoderHam — CHANGES_REQUESTED** [review]
Good work with the cleanup. Is the general convention to not have [] as a default value for params?

File: qa/common/nightly_email_helper.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
```
Comment (CoderHam) [inline qa/common/nightly_email_helper.py:39]: 'base64' is not used. You should be able to remove that import

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: qa/common/nightly_email_helper.py (hunk 1 above)
Comment (dyastremsky) [reply qa/common/nightly_email_helper.py:42]: It's because of the attachments.append. Default arguments are instantiated once, not once per call. By using .append, the default argument is changed. So the behavior would vary for every call (i.e. once a new attachment is appended when using the default value, that attachment would already be appended for the next call with the default attachments value).

File: qa/common/reporter.py [hunk 3]
```diff
@@ -139,7 +139,8 @@ def post_to_url(url, data):
```
Comment (CoderHam) [inline qa/common/reporter.py:139]: Import of 'sys' is not used.

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Comment by CoderHam** [comment]
> @dyastremsky Did you use any tool to detect the issues? Can you do something similar for Python backend?

@tanmayv25 Similar tool for python backend
https://lgtm.com/projects/g/triton-inference-server/python_backend?mode=list

We should do a similar run through the client repo

File: qa/common/nightly_email_helper.py (hunk 1 above)
Comment (CoderHam) [reply qa/common/nightly_email_helper.py:42]: TIL 😄

File: qa/common/reporter.py (hunk 3 above)
Comment (dyastremsky) [reply qa/common/reporter.py:138]: Removed.

File: qa/common/nightly_email_helper.py (hunk 2 above)
Comment (dyastremsky) [reply qa/common/nightly_email_helper.py:38]: Removed.

**Review by CoderHam — DISMISSED** [review]

**Comment by dyastremsky** [comment]
Thank you both! Addressed the feedback regarding other imports to get rid of. There will likely be more later on, as I only addressed a hundred or so of the higher-priority alerts. Didn't want to do them all at once, as I'm already trying to figure out why [rate_limiter--base is failing](https://gitlab-master.nvidia.com/dl/dgx/tritonserver/-/jobs/29411262), so figured it might make sense to split into a couple PRs.

**Review by GuanLuo — APPROVED** [review]

**Review by tanmayv25 — DISMISSED** [review]

File: qa/L0_rate_limiter/rate_limiter_test.py [hunk 4]
```diff
@@ -267,7 +263,7 @@ def test_single_model_sequence_batching(self):
 
         try:
             model_name = "custom_sequence_int32"
-            self.assertFalse("TRITONSERVER_DELAY_SCHEDULER" in os.environ)
```
Comment (tanmayv25) [inline qa/L0_rate_limiter/rate_limiter_test.py:270]: Note: this was assertFalse.. We are checking whether TRITONSERVER_DELAY_SCHEDULER was properly unset before running this test.

Comment (dyastremsky) [reply qa/L0_rate_limiter/rate_limiter_test.py:270]: Thanks for finding this!

**Review by CoderHam — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]
